### PR TITLE
Paul test local - this one has everything

### PR DESCRIPTION
**Pull request by starsinmypockets**
@boutell I made incremental PRs in case that made sense, but this branch `paul_test_local` has all of the updated test and library code, and removes the old `test.js` ad hoc. I hope this isn't confusing but this branch should pass tests for all existing storage backends (local fs, s3, and azure).

### CODE REVIEW

File: lib/storage/azure.js [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      path = path[0] === '/' ? path.slice(1) : path;
+
+      async.each(blobSvcs, function(blobSvc, cb) {
+        removeContainerBlob(blobSvc, path, cb);
+      }, callback);
+    },
+
+    disable: function(path, callback) {
+      const dPath = utils.getDisabledPath(path, self.options.disabledFileKey);
+      blobSvcs.forEach(blob => {
```
Comment (boutell) [inline lib/storage/azure.js]: This must

Comment (boutell) [reply lib/storage/azure.js]: This must use async.each, like in Ludo's versions. forEach will wind up invoking the callback several times.

Comment (boutell) [reply lib/storage/azure.js]: Use ludo's implementations as a modal here and you should be OK.

File: lib/storage/azure.js [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+          } else { // otherwise, remove original file (azure does not currently support rename operations, so we dance)
+            self.remove(path, callback);
+          }
+        });
+      });
+    },
+
+    enable: function(path, callback) {
+      const dPath = utils.getDisabledPath(path, self.options.disabledFileKey);
+      blobSvcs.forEach(blob => {
```
Comment (boutell) [inline lib/storage/azure.js]: See above

**Comment by starsinmypockets** [comment]
OK this should be good @boutell

**Pull request merged**